packages/webdriver/src/command.ts

```diff
@@ -120,10 +130,33 @@ export default function (
             body[commandParams[i].name] = arg
         }
 
-        const request = new environment.value.Request(method, endpoint, body, isHubCommand)
+        /**
+        * Masking text value when matching regEx
+        */
+        let maskedBody: Record<string, any> | undefined
+        let maskedArgs: string[] | undefined
+        if (maskingPatterns.length > 0 && commandInfo.parameters.some((param) => param.name === 'text')) {
+            const sensitiveReplacer = '*SECURE*'
+            const hasSensitiveTextData = Object.entries(body).some(([commandParam, paramValue]) => commandParam === 'text' && maskingRegExps.some((pattern) => pattern.test(paramValue)))
+            if (hasSensitiveTextData) {
+                maskedBody = {
+                    ...body,
+                    text: sensitiveReplacer
+                }
+                maskedArgs = args.map((arg) => maskingRegExps.some((pattern) => pattern.test(arg)) ? sensitiveReplacer : arg)
+            }
+        }
+
+        const request = new environment.value.Request(
+            method,
+            endpoint,
+            body,
+            maskedBody,
```

Comment:
Do we actually want to mask the inputs for the WebDriver request? What if someone wants to use actual credentials to log in but don't want to have them exposed in their logs?

Reply:
> to mask the inputs for the WebDriver request

I do not mask the request per see that is sent; I only set a masked version and use it in all other places that are not the exact request. 

This is how I masked my actual password. My password values are `@maks@adwegbefb4ty3henttn35,`, and the RegExp is `/@mask@.*/i`. I use the same RegExp to mask it from the appium log with [log filters](https://appium.io/docs/en/2.0/guides/log-filters/). This appium solution is hard to use, but I wanted to align.

Reply:
If I may - this is the exact use case @christian-bromann is describing that I've stumbled upon. Im using actual credentials, even if it's a test account, to sign-in into our Azure AAD B2C. The value is injected via env-var and simply set (.setValue) on the selected element, but is logged in clear-text on Browserstack. 
Would be nice to also have it masked on Browserstack itself.

Reply:
🤔 I can take a look; it is just that the quick fix in before-after hooks now is growing way more than I anticipated.

Reply:
@zyntax90 My approach is to align with the appium approach of [log filters](https://appium.io/docs/en/2.0/guides/log-filters/). If we diverge, you could be masking in wdio logs but not in appium ones.

Maybe I misused the appium strategy, but besides setting a known keyword in my password, I could not mask my values from the appium logs, which output them once with the value surrounded by double quotes and another time with all my password characters separated by a comma (see screenshot below). 

![image](https://github.com/user-attachments/assets/aef12e3f-2f85-4447-a0ae-a127843a019b)

My proposed option is to ensure we can be masked in both wdio & appium logs with the same RegExp.

I was not too fond of the appium strategy, but I'm all open to suggestions. Again, if we diverge, people will still need to find some ways to find their proper RegExp using the appium log filter; unless we succeed in also implementing something different in appium... we can dream 😄 

To brainstorm:
Idea #1:
- Instead of a RegExp, as I propose, that masked everything. We could add an option to remove a matching group.
So the config would be like below. `removeOnGroupMatch` could even be optional, and we could just state in the doc that matching groups or removed before sending the value to appium

```
const config = { 
    maskPattern: '/(@mask@).*/i',
    removeOnGroupMatch: true
}
```
With `@mask@myPassword` I have the below
![image](https://github.com/user-attachments/assets/d5e7030c-a33e-4f1f-b0e9-dc6d3ef14f47)

Then, I would replace Group 1 with an empty string and send that to appium.

For appium log now, that cannot apply, and we will need to disable the appium logging!

Idea #2 .... thinking!

Reply:
Also @zyntax90 , for
> but is logged in clear-text on Browserstack.
You need to use [maskCommand](https://www.browserstack.com/docs/automate/selenium/hide-sensitive-data) else what ever we do in wdio it will never mask their `Text Logs`

Reply:
> Also @zyntax90 , for
> 
> > but is logged in clear-text on Browserstack.
> 
> You need to use [maskCommand](https://www.browserstack.com/docs/automate/selenium/hide-sensitive-data) else what ever we do in wdio it will never mask their `Text Logs` (as far as I know)

Ah, thanks for the clarification and I do think it's a neat idea of using matching group.

Reply:
@zyntax90 with the appium log disabled in that case, right?